lib/open_food_network/products_renderer.rb

```diff
@@ -27,25 +29,33 @@ def products_json
 
     private
 
+    attr_reader :order_cycle, :distributor, :customer, :params
+
     def products
-      return unless @order_cycle
+      return unless order_cycle
 
       @products ||= begin
-        scoper = ScopeProductToHub.new(@distributor)
+        results = distributed_products.products_relation.order(taxon_order)
 
-        distributed_products.products_relation.
-          order(taxon_order).
-          each { |product| scoper.scope(product) }
+        filter_and_paginate(results).each { |product| product_scoper.scope(product) }
```

Comment:
I think we need a little comment here explaining to the future dumb devs (ourselves next year :-)) looking at this code. Something like "distributed_products is already taking Overrides into account in the query, here we need to scope the products so that the overrides are taken into account on the serialization process that follows"
and btw, is this explanation correct?